data/virt_autotest/virt_logs_collector.sh

```diff
@@ -0,0 +1,427 @@
+	shift
+	shift
+	local extra_logs=$@
+        collect_logs_via_guest_console ${guest_domain} ${logs_folder} ${extra_logs}
+        if [[ $? -eq 0 ]];then
+           return 0
+        else
+           return 1
+        fi
+}
+
+#Collect any extra logs wanted from guest. Will resort to guest virsh console if collecting from ssh failed.
+function collect_extra_logs_from_guest() {
+        local logs_folder=$1
+        local guest_ipaddr=$2
+        local guest_domain=$3
+        shift
+        shift
+        shift
+        local extra_logs=$@
+      
+	if [[ ${extra_logs} != "" ]];then
+           local guest_user="root"
+           local guest_pass="novell"
+           local sshpass_ssh_cmd="sshpass -p ${guest_pass} ssh ${guest_user}@${guest_ipaddr}"
+           local guest_transformed=${guest_domain//./_}
+           local ret_result=128
+           local retry_times=0
+           while [[ ${retry_times} -lt 3 ]] && [[ ${ret_result} -ne 0 ]];
+           do
+                 ${sshpass_ssh_cmd} mkdir -p ${logs_folder}/guest_${guest_transformed}_extra_logs
+                 echo -e "${sshpass_ssh_cmd} cp -r -f ${extra_logs} ${logs_folder}/guest_${guest_transformed}_extra_logs"
+                 ${sshpass_ssh_cmd} cp -r -f ${extra_logs} ${logs_folder}/guest_${guest_transformed}_extra_logs
+                 ret_result=$?
+                 if [[ ${ret_result} -eq 0 ]];then
+                    echo -e "Successfully collected ${extra_logs} from guest ${guest_domain} via ssh."
+                    break
+                 fi
+                 retry_times=$((${retry_times}+1))
+           done
+           if [[ ${ret_result} -ne 0 ]];then
+              echo -e "Can not collect ${extra_logs} from guest ${guest_domain} via ssh. Try to use guest virsh console"
+              echo -e "collect_extra_logs_via_guest_console ${guest_domain} ${logs_folder} ${extra_logs}"
+              collect_extra_logs_via_guest_console ${guest_domain} ${logs_folder} ${extra_logs}
+	      if [[ $? -eq 0 ]];then
+                 return 0
+              else
+                 return 1
+              fi
+           else
+	      return 0
+	   fi
+        else
+           echo -e "There is no extra logs to be collected. Please specify something."
+	   return 0
+        fi
+}
+
+#Collect any extra wanted logs from host. And provide more complete virtualization logs for SLE-11-SP4 and SLE-12 hosts.
+function collect_extra_logs_from_host() {
+	local logs_folder=$1
+	shift
+	local extra_logs=$@
+	local ret_result=0
+
+	if [[ ${extra_logs} != "" ]];then
+	   cp -f -r ${extra_logs} ${logs_folder}
+           ret_result=$?
+	   if [[ ${ret_result} -eq 0 ]];then
+	      echo -e "Successfully collected extra logs ${extra_logs} from host."
+	   else
+	      echo -e "Failed to collect extra logs ${extra_logs} from host."
+	   fi
+	else
+	   echo -e "No extra logs to be collected from host. Please specify something."
+	fi
+
+	local release=`get_sles_release release`
```

Comment:
I understand that adding the logs below will supplement the missing virt logs. But putting them in this function will break the freedom of this function when users just want to add the customized logs  that they want.

Reply:
This option adds flexibility instead of "breaking something". You can just leave them empty if do not want to use. So easy.  And this option just supports "customized logs".

Reply:
Seems we are not talking about the same thing. I do not mean the option thing. I mean line 288~315. Putting them in this function will break the freedom of this function when users just want to add the customized logs that they want. For example, when users just want to add extra log ~/customized, these lines will collected the unneeded libvirt/xen logs.

Reply:
For host yes, those SLES host needs this extra virtualization logs because they are lacking of them. This is by design.